xwiki-platform-core/xwiki-platform-web/src/main/webapp/templates/shareinline.vm

```diff
@@ -25,10 +25,11 @@
       #set ($sender = "$!{escapetool.q($sender)} <noreply@$!{serverName}>")
     #end
   #end
-  #set ($recipients = [])
+  #set ($recipients = $collectionstool.set)
   #set ($errors = {})
   #set ($successes = [])
 
+  #set ($templateReference = $services.model.createDocumentReference('', 'XWiki', 'SharePage'))
   #foreach ($recipientValue in $request.getParameterValues('target'))
     #foreach ($recipient in $recipientValue.split(','))
```

Comment:
I think this should be refactored to use https://extensions.xwiki.org/xwiki/bin/view/Extension/Mail%20Sender%20API#HExample4:SendaTemplateemailtoalistofUsersandGroups since this s a feature supported by the Mail Send API module and it's more performant, using iterators. It should also allow deduplicating and not send email several times to the same person (if they're part of several groups for ex).

Reply:
This functionality is not yet covered by share by email & will be implemented in a future commit/PR.